### ISSUE TITLE
PowerShell Integrated Console appears to start twice

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/vscode-powershell#604]
### System Details

- Operating system name and version: Windows 10 AU
- VS Code version: 1.10.2
- PowerShell extension version: 0.11.0
- Output from `$PSVersionTable`:

```
Name                           Value
----                           -----
PSVersion                      5.1.14393.953
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.953
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Issue Description

When I open a PS workspace in VSCode the first time, the integrated console appears to load twice:
![pses-ic-startup](https://cloud.githubusercontent.com/assets/5177512/24228328/15220970-0f39-11e7-96a8-9708be942107.gif)
I also notice that the first time, it displays the logo but then the screen is cleared and it says it's starting `PowerShell` again.  Is this the new host that loads the second time?  If so, maybe it could say `Starting the PowerShell Integrated Console...`?  If we can't prevent the first (original) load, we may want to see if we can start PowerShell with `-NoLogo`.  For the final load, do we want to show a logo?  Perhaps with a setting to disable it?

### Attached Logs

N/A I think.  If that's not the case, let me know and I can attach them.

**Comment by daviwil** [comment]
VS Code is doing this, here's the issue I filed against them about it: https://github.com/Microsoft/vscode/issues/22968

Basically, when VS Code restarts, it remembers that **a** terminal was open when the window was closed the previous time and then launches the terminal UI using your default terminal preference just before I have a chance to start the integrated console.  I have tried a few different ways of suppressing it but nothing has worked yet.

**Comment by jmiller76** [comment]
Wouldn't a better process be to set something in terminal.integrated.shell.windows that would start the extension and the integrated terminal? 

This would set a user setting that would prevent accidently killing whatever other terminal may or may not be wanted.  It would also be more efficient if the default powershell is loading with a profile.

**Comment by JustinGrote** [comment]
Now that https://github.com/Microsoft/vscode/issues/13267 is closed, can we revisit this? It is super annoying still :)

**Comment by rjmholt** [comment]
It looks like this is no longer occurring. Closing this now, but can reopen if needed

**Comment by JustinGrote** [comment]
Can confirm, I just got so used to it happening that I never stopped to notice when it didn't happen. Cool beans!

**Comment by JustinGrote** [comment]
@rjmholt @TylerLeonhardt Actually this is still an issue at least with "Reload Window" and should be reopened, I just didn't look closely enough.

### Steps to reproduce:
1. Start window with Powershell Integrated Terminal
2. Choose "Reload Window"
3. A pwsh terminal in addition to integrated terminal will be started.

### Expected behavior
Only the Powershell Integrated Terminal should be started upon a reload window if it was the only terminal open previously.
![Capture](https://user-images.githubusercontent.com/15258962/76997494-9b0dfd80-6910-11ea-9b99-3e84a3a54557.gif)

**Comment by TylerLeonhardt** [comment]
ah I see... that's because the Terminal pane is open and by default it opens your default shell...

**Comment by JustinGrote** [comment]
> ah I see... that's because the Terminal pane is open and by default it opens your default shell...

Correct, in the past there was nothing you could do about it, but you should be able to now with https://github.com/microsoft/vscode/issues/13267

**Comment by TylerLeonhardt** [comment]
Yeah but... I don't know if we want to mess with other terminals... could be dangerous. The user might have been already running something in that other terminal.

**Comment by JustinGrote** [comment]
Related issue that was abandoned.
https://github.com/microsoft/vscode/issues/39137

Here's a workaround script maybe that can be integrated
https://github.com/microsoft/vscode/issues/39137#issuecomment-414127477

**Comment by aproposts** [comment]
This is still feels like it should be fixed by the vscode team. The argument that they're restoring state doesn't track: If I didn't have a `pwsh` (or whatever) open in the Terminal tab when I closed the window, why is it being 'restored'?

We're not the only ones [complaining about this](https://github.com/microsoft/vscode/issues/147192). I don't know if a new settings option would even be necessary: If I have both of the persist settings set as such:

```json
  "terminal.integrated.persistentSessionReviveProcess": "never",
  "terminal.integrated.enablePersistentSessions": false,
```

...it's pretty clear I don't want the application opening a terminal for me. Regardless, I'll try to find some time to carefully read the open issue and make a persuasive argument.

**Comment by andschwa** [comment]
We would definitely appreciate that, requests come best from the users themselves!

**Comment by andyleejordan** [comment]
This appears to have been addressed in **v2023.6.0** (June 2023), which refactored the client startup logic so a ghost Extension Terminal can't be created, and [#4599](https://github.com/PowerShell/vscode-powershell/pull/4599) made `sessionManager.start()` idempotent so the session can't start twice. If you still see this on a current version, please comment with fresh repro steps and logs and we'll reopen.

<sub>🤖 Drafted by GitHub Copilot at @andyleejordan's request — pending maintainer review.</sub>
